Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -37,6 +37,7 @@ void GridMasonryLayout::performMasonryPlacement(unsigned gridAxisTracks, GridTra
     m_masonryAxisDirection = masonryAxisDirection;
     m_masonryAxisGridGap = m_renderGrid.gridGap(m_masonryAxisDirection);
     m_gridAxisTracksCount = gridAxisTracks;
+    m_gridContentSize = 0;
```

Comment:
You should add a comment, like was done for `m_autoFlowNextCursor` to explain why you are clearing this.

Better still would be a helper function, `resetMasonryState` or something that clears both values.

Reply:
I'll do a follow up patch to make a new helper function.